e2e/policy/policy_test.go

```diff
@@ -0,0 +1,175 @@
+		require.NoError(c.WaitFor(ctx, kubeclient.Deployment{}, ct.Namespace, opensslFrontend))
+
+		time.Sleep(5 * time.Second) // let the error counter go up initially
+
+		// get the attestation failures before removing a policy
+		initialFailures := getFailures(ctx, t, ct)
+
+		t.Log("Initial failures:", initialFailures)
+
+		// parse the manifest
+		manifestBytes, err := os.ReadFile(path.Join(ct.WorkDir, "manifest.json"))
+		require.NoError(err)
+		var m manifest.Manifest
+		require.NoError(json.Unmarshal(manifestBytes, &m))
+
+		// Remove a policy from the manifest.
+		newPolicies := make(map[manifest.HexString][]string)
+		for policyHash := range m.Policies {
+			if slices.Contains(m.Policies[policyHash], opensslFrontend) {
+				continue
+			}
+			newPolicies[policyHash] = m.Policies[policyHash]
+		}
+		m.Policies = newPolicies
+
+		// write the new manifest
+		manifestBytes, err = json.Marshal(m)
+		require.NoError(err)
+		require.NoError(os.WriteFile(path.Join(ct.WorkDir, "manifest.json"), manifestBytes, 0o644))
+
+		// parse the original resources
+		resourceBytes, err := os.ReadFile(path.Join(ct.WorkDir, "resources.yaml"))
+		require.NoError(err)
+		r, err := kubeapi.UnmarshalUnstructuredK8SResource(resourceBytes)
+		require.NoError(err)
+
+		// remove everything from the openssl-frontend
+		newResources := r[:0]
+		for _, resource := range r {
+			name := resource.GetName()
+			require.NoError(err)
+			if name == opensslFrontend || name == "port-forwarder-"+opensslFrontend {
+				continue
+			}
+			newResources = append(newResources, resource)
+		}
+
+		// write the new resources yaml
+		r = newResources
+		resourceBytes, err = kuberesource.EncodeUnstructured(r)
+		require.NoError(err)
+		require.NoError(os.WriteFile(path.Join(ct.WorkDir, "resources.yaml"), resourceBytes, 0o644))
+
+		// set the new manifest
+		ct.Set(t)
+
+		// restart the deployments
+		require.NoError(c.Restart(ctx, kubeclient.Deployment{}, ct.Namespace, opensslFrontend)) // not waiting since it would fail
+		require.NoError(c.Restart(ctx, kubeclient.Deployment{}, ct.Namespace, opensslBackend))
+		require.NoError(c.WaitFor(ctx, kubeclient.Deployment{}, ct.Namespace, opensslBackend))
+
+		// wait a bit to let the attestation failure counter go up
+		time.Sleep(5 * time.Second)
+
+		newFailures := getFailures(ctx, t, ct)
+		t.Log("New failures:", newFailures)
+		// errors should happen
+		require.Greater(newFailures, initialFailures)
+	})
+}
+
+func TestMain(m *testing.M) {
+	flag.StringVar(&imageReplacementsFile, "image-replacements", "", "path to image replacements file")
+	flag.StringVar(&namespaceFile, "namespace-file", "", "file to store the namespace in")
+	flag.BoolVar(&skipUndeploy, "skip-undeploy", false, "skip undeploy step in the test")
+	flag.Parse()
+
+	os.Exit(m.Run())
+}
+
+func parsePrometheus(input string) (map[string]*dto.MetricFamily, error) {
+	var parser expfmt.TextParser
+	return parser.TextToMetricFamilies(strings.NewReader(input))
+}
```

Comment:
```suggestion
func parsePrometheus(input string) (map[string]*dto.MetricFamily, error) {
	return expfmt.TextParser{}.TextToMetricFamilies(strings.NewReader(input))
}
```

... and then inline the whole thing?